src/skin.cpp

```diff
@@ -42,6 +42,7 @@ void generateSkinAreas(int layer_nr, SliceMeshStorage& mesh, const int innermost
         }
 
         Polygons upskin = part.insets.back().offset(-innermost_wall_line_width / 2);
+        Polygons virgin_upskin = Polygons(upskin);
```

Comment:
It's really just the original skin outline which gets used to initialize upskin (and downskin). I have now renamed virgin_upskin to original_outline.